providers/base/tests/test_pipewire_test.py

```diff
@@ -0,0 +1,364 @@
+# Copyright 2023 Canonical Ltd.
+# Written by:
+#   Hanhsuan Lee <hanhsuan.lee@canonical.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+import sys
+from unittest.mock import MagicMock
+from unittest.mock import Mock, patch, mock_open, call
+try:
+    sys.modules["gi"] = MagicMock()
+    sys.modules["gi.repository"] = MagicMock()
+    from pipewire_test import *
+except ImportError:
+    sys.exit(127)
```

Comment:
I thought unit test is to test the function and not affect by the difference of machine, no matter who or what running the test. should come out the same result. Therefore I wonder what's the rationale of `try/except` here

Reply:
I just copied from the `pipewire_test.py`. Sure let me remove it.

Reply:
NO, you shouldn't just copy the code from `audio_test.py` and mofidy a few of the content to create another script. Please don't ever do this, coding like this will make the whole checkbox fat and redundant. You can either add your function into it or use `decorator` to create another one if you prefer.

Reply:
My thought is that `audio_test.py` should be retired someday, then your approach will make checkbox fat and redundant.

Reply:
Are you saying that all the ubuntu version will retire pulseaudio? including uc18, 20, 18.04, 20.04?

Reply:
I mean after 22.04 is retired. Reusing code is great, but I do think your approach will create another monster `udevadm.py` in the future for compatibility issue.

Reply:
Main route for a follow up, a de-duplication of this module, resolving for now